File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let answer = 0;
    
    let start = 0, end = 9;
    while (end <= discount.length-1) {
        let checkNumber = new Array(number.length).fill(0);
        
        for (let i = start; i <= end; i++) {
            for (let j = 0; j < want.length; j++) {
                if (discount[i] === want[j]) {
                    checkNumber[j]++;
                }
            }
        }
                
        let flag = true;
        for (let i = 0; i < checkNumber.length; i++) {
            if (checkNumber[i] < number[i]) {
                flag = false;
                break;
            }
        }
        if (flag) {
            answer++;
        }
        start++;
        end++;        
    }
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def check(want, number, dict):
    
    for i in range(len(want)):
        if want[i] not in dict or dict[want[i]] < number[i]:
            return False
    
    return True
    

def solution(want, number, discount):
    answer = 0
    
    dict = {}
    
    for i in range(10):
        if discount[i] not in dict:
            dict[discount[i]] = 1
        else:
            dict[discount[i]] += 1
            
    if check(want, number, dict):
        answer += 1
    
    for i in range(10, len(discount)):        
        dict[discount[i-10]] -= 1
        
        if discount[i] in dict:
            dict[discount[i]] += 1
        else:
            dict[discount[i]] = 1
            
        if check(want, number, dict):
            answer += 1
        
        
    return answer